README.md
# Murder Mysetry
You're a detective who must solve a murder case in order to save your own life. 

## Installation
Python 3.7 is needed to play this game.

## How To Run The Program

python game.py

## Collaboration

Feedback and criticism are welcome. 

For major changes, please open an issue first to discuss what you would like to change.

Please make sure to update tests as appropriate.


changelog.md
# Change Log 
Includes all important changes to the RPG adventure game.

**v0.3 - Prerelease**
- Added maps v0.1
- Added Continuous Game Play
- Added RPG classes (characters)

**v0.2 - Prerelease**
- Added pseudocode v0.1
- Updated game.py file to have numerical lists

**v0.1 - Prerelease**
- Added ReadMe Section describing the project.
- Added Change Log section for important modifications.


game.py
# Course: CS 30
# Period: 1
# Date created: 20/09/29
# Date last modified: 20/09/30
# Name: Brahm Gainda
# Description: RPG Nested Dictonaries


# Dictonary for characters

characters = {1: {'name': 'Barb', 'age': '21', 'sex': 'Male',
              'relevance': 'Detective', 'trait': 'magnificent at theyre work'},
              2: {'name': 'bill', 'age': '24', 'sex': 'Male',
              'relevance': 'Suspect', 'trait': 'friendly'},
              3: {'name': 'sam', 'age': '19', 'sex': 'Male',
              'relevance': 'Suspect', 'trait': 'short-tempered'},
              4: {'name': 'Steven', 'age': '28', 'sex': 'Male',
              'relevance': 'Suspect', 'trait': 'quiet'},
              5: {'name': 'Shaniqua', 'age': '18', 'sex': 'Female',
              'relevance': 'Suspect', 'trait': 'manipulative'},
              6: {'name': 'Timmy', 'age': '31', 'sex': 'Male',
              'relevance': 'extra Suspect', 'trait': 'impulsive'},
              7: {'name': 'Chad', 'age': '23', 'sex': 'Male',
              'relevance': 'extra Suspect', 'trait': 'short-tempered'}}

for character, character_info in characters.items():
    name = character_info['name'].title()
    age = character_info['age']
    sex = character_info['sex']
    relevance = character_info['relevance']
    trait = character_info['trait']

    print("\n" + name.title() + " is a " + age + " year old " + sex)
    print(" They are a " + relevance + " in the case.")
    print(" They are considerably " + trait + " .")

# Dictionary For Inventory

print('\n\n')

inventory = []

clue = {
    'item': 'Bills DNA',
    'description': 'clue found at the crime scene',
    'persuasion_points': 2,
    }
inventory.append(clue)

clue = {
    'item': 'Bills shoe',
    'description': 'a clue found at the crime scene',
    'persuasion_points': 2,
    }

inventory.append(clue)

clue = {
    'item': 'Sams hair',
    'description': 'extra clue that was hidden at the crime \
scene',
    'persuasion_points': 4,
    }
inventory.append(clue)

clue = {
    'item': 'a gun with Bills finger prints',
    'description': 'extra clue at the \
crime scene',
    'persuasion_points': 4,
    }
inventory.append(clue)


for inventory in inventory:
    print("\n you found " + inventory['item'])
    print(" thats a " + inventory['description'] + ".")
    print(" It has " + str(inventory['persuasion_points']) +
          " persuasion points.")

# Dictionary For Locations and Descriptions

print('\n\n')

locations = []

description = {
    'location_name': 'New York City',
    'description': ' you and the deadly gang live, and where the game is set.',
    }
locations.append(description)

description = {
    'location_name': 'Chuck E Cheese',
    'description': ' the murder occurred and where you look for \
clues',
    }
locations.append(description)


description = {
    'location_name': 'An Abandoned Warehouse on the outskirts of the city',
    'description': ' the deadly gang that has \
kidnapped you lives.',
    }
locations.append(description)

description = {
    'location_name': 'A room in the abandoned warehouse',
    'description': ' you interrogate the suspects.',
    }
locations.append(description)

for location in locations:
    print("\n" + location['location_name'])
    print("is where" + location['description'])


maps.py
# Course: CS 30
# Period: 1
# Date created: 20/10/16
# Date last modified: 20/10/19
# Name: Brahm Gainda
# Description: RPG modules and maps

from tabulate import tabulate

# defining map


def print_map():
    """Function for printing grid map of Murder Mystery game"""
    map = [['', 'Interrogation Room'],
           ['Empty Fields   ', 'The Gangs House', 'Empty Fields'],
           ['', 'Crime Scene', ]]
    print(tabulate(map, tablefmt="grid"))


# creating a class for the rooms in the map
class Map:
    """Class for different rooms and their relevance"""
    def __init__(self, room, relevance):
        self.room = room
        self.relevance = relevance

    def __str__(self):
        desc = (f"the {self.room} is the room where "
                f"{self.relevance}")
        return desc

# Defining the different rooms
interrogation_room = Map("Interrogation Room", "you interrogate suspects to\
see if their guilty")
the_gangs_house = Map("The Gangs House", "you are being kept\
prisoner till you solve the case")
crime_scene = Map("Crime Scene", "is where the murder occurred\
and where you look for clues")


characters.py
# Course: CS 30
# Period: 1
# Date created: 20/10/19
# Date last modified: 20/10/23
# Name: Brahm Gainda
# Description: RPG Classes

from tabulate import tabulate

# Classes For Suspects


class Suspect:
    """Parent class for primary/secondary suspects"""
    def __init__(self, name, trait):
        self.name = name
        self.trait = trait


class PrimarySuspect(Suspect):
    """Class for primary suspects"""
    def __init__(self, name, trait):
        super().__init__(name, trait)

    def __str__(self):
        desc = (f"{self.name} is very {self.trait}\n"
                f"{self.name} is a primary suspect, theres a\
                high chance they committed the crime...")
        return desc


class SecondarySuspect(Suspect):
    """Class for secondary suspects"""
    def __init__(self, name, trait):
        super().__init__(name, trait)

    def __str__(self):
        desc = (f"{self.name} is very {self.trait}\n"
                f"{self.name} is a secondary suspect, theres a\
                lower chance they committed the crime...")
        return desc

# Defining the different suspects
bill = PrimarySuspect("Bill", "high tempered")
shaniqua = PrimarySuspect("Shaniqua", "manipulative")
steven = PrimarySuspect("Steven", "shy")
sam = PrimarySuspect("Sam", "high tempered")
timmy = SecondarySuspect("Tim", "loud")
chad = SecondarySuspect("Chad", "fidgety")


# Map Class
def print_map():
    """Function for printing grid map of Murder Mystery game"""
    map = [['', 'Interrogation Room'],
           ['Empty Fields   ', 'The Gangs House(start)', 'Empty Fields'],
           ['', 'Crime Scene', 'Police Station(end)']]
    print(tabulate(map, tablefmt="grid"))


# creating a class for the rooms in the map
class Map:
    """Class for different rooms and their relevance"""
    def __init__(self, room, relevance):
        self.room = room
        self.relevance = relevance

    def __str__(self):
        desc = (f"the {self.room} is the room where "
                f"{self.relevance}")
        return desc

# Defining the different rooms
interrogation_room = Map("Interrogation Room", "you interrogate suspects to\
see if their guilty")
the_gangs_house = Map("The Gangs House", "you are being kept\
prisoner till you solve the case")
crime_scene = Map("Crime Scene", "is where the murder occurred\
and where you look for clues")
empty_fields = Map("Empty Fields", "you go to think in silence")

# asks the user where they want to go on map
message = "Print 'map' to see map\
Where would you like to go first?\
Interrogation Room, The Gangs House, Crime Scene, Empty Fields "

keypress = input(message)

# where the user goes when they input
if keypress.lower() == "map":
    print_map()
if keypress.lower() == "interrogation room":
    print(interrogation_room)
if keypress.lower() == "the gangs house":
    print(the_gangs_house)
if keypress.lower() == "crime scene":
    print(crime_scene)
if keypress.lower() == "empty fields":
    print(empty_fields)


pseudocode.py


# Start game by giving player backstory
# define movement variables
# define interaction variables
# define useuables  variables
# display list of suspects and clues

# give player option to investigate a suspects
# give player option to move one or keep investigating
# if player investigates more give option to tell the gang he solved the case
# if player moves one give him option to investigate another suspects
# Also if  player moves on give option to re investigate the crime scence 
# if player chooses suspect, give option to move on or say they solved the case
# if player chooses crime scence, tell player they found a new clue
# Also if player chooses crime scence give option to keep searching

# player may investigate each suspect once 
# game is won when player chooses the right person as the murderer

continuous_game.py
# Course: CS 30
# Period: 1
# Date created: 20/10/7
# Date last modified: 20/10/21
# Name: Brahm Gainda
# Description: RPG Continuous Game Play

# Empty Lists Used To Add Iteams Into
Suspects = []
Clues = []
Persuasion_points = {}

# Prompts Used as User Input
interrogate = "Who would you like to interrogate first?\n> "

clues_message = "Would you like to search for more (yes/no)?"

extraclue_found = "You found an extra clue 'Sams hair at the crime scene'\
= 5 p.p. against Sam"

extraclue2_found = "You found an extra clue 'Gun with Bills finger prints'\
= 2 p.p. against Bill"

message = "What would you like to do first?\
Check Suspects List, Check Clues, Check persuasion points"

# Intro Messages Explaining The Game
print("enter 'quit' at any time to quit ")
print("\nYou're a famouse detective who has been kidnapped by a deadly gang.")
print("They tell you that a member of they're group has been \
murdered and they want you to solve the case\n")
print("They give you 24 hours to solve the case \
and a list of suspects and clue")
print("to convince the gang you solved the case \
you have to find 10 persuasion points against a suspect")
print("\nso find clues and interrogate suspects \
to get persuasion points against suspects \
and solve the case so you're life is spared")
print("\nWhat would you like to do first?")

# Fixed Game Info That Is Used Throughout The Code
suspects_list = ['Bill', 'Shaniqua', 'Steven', 'Sam']

clues = ['Bills shoe at crime scene = 2 p.p.', 'Bills DNA at crime scene\
= 2 p.p']

persuasion_points = {
                      'bill': 4,
                      'shaniqua': 0,
                      'steven': 0,
                      'sam': 0
                    }

hours_left = 24

# Functions Used Throughout The Game


def game_input(message):
    "Ends the program if the user inputs 'quit' "
    choice = input(message)
    if choice.lower() == "quit":
        exit()
    return choice


def investigation(hours_left):
    "Returns The # of turns the user has left in the game"
    while hours_left > 0:
        message = """
Check Suspects List, Check Clues, Check Persuasion Points
(You have {} hours left)
> \
""".format(hours_left)
        # Prompt That Asks The User Their Desired Path
        x = game_input(message)
        # Suspects path, One Path The User Could Use
        if x.lower() == "check suspects list":
            print('\nSuspects: ' + ', '.join(suspects_list))
            choice = game_input(interrogate)
            if choice.lower() in Suspects:
                print("You've already interrogated {}".format(choice))
                continue
            if choice.lower() == "bill" or \
               choice.lower() == "shaniqua" or \
               choice.lower() == "steven" or \
               choice.lower() == "sam":
                    hours_left -= 2
                    Suspects.append(choice)
                    print(f"\nYou interrogated {choice}, they messed up \
and you have 2 persuasion points againt them")
                    persuasion_points[choice] += 2
                    print("you have " + str(hours_left) + " hours left")
                    break
            else:
                print("\nThat isn't a suspect")
                continue

                # Clues Path, A Path The User Could Choose
        if x.lower() == "check clues":
            print("\nHere are the clues you have:")
            print(clues)
            print("\n")
            choice = game_input(clues_message)
            if choice.lower() in clues:
                print(f"You've already found {choice}")
                break
            if choice.lower() == "yes":
                    print("\n" + extraclue_found)
                    persuasion_points['sam'] += 4
                    hours_left -= 4
                    print("you have " + str(hours_left) + " hours left")
                    break
            if choice.lower() == "no":
                break
            else:
                print("That isn't a choice")
                continue

        # Persuasion Points Path, Another Path The User Could Choose
        if x.lower() == "check persuasion points":
            print("\nHere are the persuasion points you have:")
            print(persuasion_points)
            print("\n")
            choice = game_input(clues_message)
            if choice.lower() == "yes":
                    print("\n" + extraclue2_found)
                    persuasion_points['bill'] += 2
                    hours_left -= 4
                    print("you have " + str(hours_left) + " hours left")
                    break
            else:
                print("That isn't a choice")
                continue

            # Quit, Incase The User Wants To Stop The Program
            if x.lower() == "quit":
                hours_left = 0
                exit()
            # Incase The User Inputs An Unknown Input
        else:
            print("\nYou did not pick one of the available options")
            print("try again")
            continue

investigation(hours_left)


c.game.py
# Course: CS 30
# Period: 1
# Date created: 20/10/7
# Date last modified: 20/10/21
# Name: Brahm Gainda
# Description: RPG Continuous Game Play

from menu.py import menu
from characters.py import Suspect

# Empty Lists Used To Add Iteams Into
Suspects = []
Clues = []
Persuasion_points = {}

# Prompts Used as User Input
interrogate = "Who would you like to interrogate first?\n> "

clues_message = "Would you like to search for more (yes/no)?"

extraclue_found = "You found an extra clue 'Sams hair at the crime scene'\
= 5 p.p. against Sam"

extraclue2_found = "You found an extra clue 'Gun with Bills finger prints'\
= 2 p.p. against Bill"

message = "What would you like to do first?\
Check Suspects List, Check Clues, Check persuasion points"

# Intro Messages Explaining The Game
print("enter 'quit' at any time to quit ")
print("\nYou're a famouse detective who has been kidnapped by a deadly gang.")
print("They tell you that a member of they're group has been \
murdered and they want you to solve the case\n")
print("They give you 24 hours to solve the case \
and a list of suspects and clue")
print("to convince the gang you solved the case \
you have to find 10 persuasion points against a suspect")
print("\nso find clues and interrogate suspects \
to get persuasion points against suspects \
and solve the case so you're life is spared")
print("\nWhat would you like to do first?")

# Fixed Game Info That Is Used Throughout The Code
suspects_list = ['Bill', 'Shaniqua', 'Steven', 'Sam']

clues = ['Bills shoe at crime scene = 2 p.p.', 'Bills DNA at crime scene\
= 2 p.p']

persuasion_points = {
                      'bill': 4,
                      'shaniqua': 0,
                      'steven': 0,
                      'sam': 0
                    }

hours_left = 24

# Functions Used Throughout The Game


def game_input(message):
    "Ends the program if the user inputs 'quit' "
    choice = input(message)
    if choice.lower() == "quit":
        exit()
    return choice


def investigation(hours_left):
    "Returns The # of turns the user has left in the game"
    while hours_left > 0:
        message = """
Check Suspects List, Check Clues, Check Persuasion Points
(You have {} hours left)
> \
""".format(hours_left)
        # Prompt That Asks The User Their Desired Path
        x = game_input(message)
        # Suspects path, One Path The User Could Use
        if x.lower() == "check suspects list":
            print('\nSuspects: ' + ', '.join(suspects_list))
            choice = game_input(interrogate)
            if choice.lower() in Suspects:
                print("You've already interrogated {}".format(choice))
                continue
            if choice.lower() == "bill" or \
               choice.lower() == "shaniqua" or \
               choice.lower() == "steven" or \
               choice.lower() == "sam":
                    hours_left -= 2
                    Suspects.append(choice)
                    print(f"\nYou interrogated {choice}, they messed up \
and you have 2 persuasion points againt them")
                    persuasion_points[choice] += 2
                    print("you have " + str(hours_left) + " hours left")
                    break
            else:
                print("\nThat isn't a suspect")
                continue

                # Clues Path, A Path The User Could Choose
        if x.lower() == "check clues":
            print("\nHere are the clues you have:")
            print(clues)
            print("\n")
            choice = game_input(clues_message)
            if choice.lower() in clues:
                print(f"You've already found {choice}")
                break
            if choice.lower() == "yes":
                    print("\n" + extraclue_found)
                    persuasion_points['sam'] += 4
                    hours_left -= 4
                    print("you have " + str(hours_left) + " hours left")
                    break
            if choice.lower() == "no":
                break
            else:
                print("That isn't a choice")
                continue

        # Persuasion Points Path, Another Path The User Could Choose
        if x.lower() == "check persuasion points":
            print("\nHere are the persuasion points you have:")
            print(persuasion_points)
            print("\n")
            choice = game_input(clues_message)
            if choice.lower() == "yes":
                    print("\n" + extraclue2_found)
                    persuasion_points['bill'] += 2
                    hours_left -= 4
                    print("you have " + str(hours_left) + " hours left")
                    break
            else:
                print("That isn't a choice")
                continue

            # Quit, Incase The User Wants To Stop The Program
            if x.lower() == "quit":
                hours_left = 0
                exit()
            # Incase The User Inputs An Unknown Input
        else:
            print("\nYou did not pick one of the available options")
            print("try again")
            continue

investigation(hours_left)
